### PR TITLE
Deprecate IErrorBase.sequenceNumber

### PR DESCRIPTION
**Pull request by markfields**
This field was used for logging and this was probably not the right abstraction for it to live in.

We can stop using it because the only places it was set have been updated to log not just sequenceNumber
but a large number of useful properties off the offending message, via `CreateProcessingError`.

Note that `ContainerClose` event will now have a field `lastSequenceNumber` which will be from `DeltaManager`, and in cases where an op processing error occurred, `messageSequenceNumber` will have the same value `sequenceNumber` used to.  See [here](https://github.com/microsoft/FluidFramework/blob/c32fff7c6214a8562bb6cf416eed94a6eb502a4f/packages/loader/container-utils/src/error.ts#L151) for the full list of op properties logged.

### CODE REVIEW

File: common/lib/common-definitions/BREAKING.md [hunk 1]
```diff
@@ -1,8 +1,14 @@
 ## 0.21 Breaking changes
 
 - [supportsTags deprecated](#supportstags-deprecated)
+- [IErrorBase.sequenceNumber deprecated](#IErrorBase.sequenceNumber-deprecated)
```
Comment (markfields) [inline common/lib/common-definitions/BREAKING.md:4]: Oops wrong package BREAKING.md

**Comment by markfields** [comment]
For the record, #2303 is the PR that introduced this property, before `DataProcessingError` came around.

File: common/lib/container-definitions/src/error.ts [hunk 2]
```diff
@@ -39,7 +39,10 @@ export interface IErrorBase {
      */
     readonly errorType: string;
     readonly message: string;
-    /** Sequence number when error happened */
+    /**
+     * Sequence number when error happened
+     * @deprecated - Was used for logging and is no longer needed due to DataProcessingError/DataCorruptionError
```
Comment (vladsud) [inline common/lib/container-definitions/src/error.ts:44]: I'd guess that it's better to mark it deprecated in 0.39 and simply remove it in main

**Review by vladsud — APPROVED** [review]

File: common/lib/container-definitions/src/error.ts (hunk 2 above)
Comment (markfields) [reply common/lib/container-definitions/src/error.ts:44]: I'll add it to a list of 0.39 changes I may make, and will update this PR to just remove it.

Comment (markfields) [reply common/lib/container-definitions/src/error.ts]: #6839

**Pull request merged**